WIP fixed

sandsmark opened the pull request:
included the english translation, ported away from directx stuff (uses sfml for everything), switched to embedding the resources (images and sounds) so it doesn't have to be run from the bin directory, and some other stuff (translating stuff in the source from german to english etc.)

seems to work well enough here on linux, but it's still a bit messy

sandsmark committed:
remove hardcoded msvc crap

sandsmark committed:
cmake

sandsmark committed:
add miniwin from devilution

sandsmark committed:
builds without windows but does not link

sandsmark committed:
add extracted english translation

sandsmark committed:
translate constants++

sandsmark committed:
translate main types

sandsmark committed:
more translate

sandsmark committed:
astyle

sandsmark committed:
some bug fix

sandsmark committed:
start porting rendering to SFML

sandsmark committed:
it actually builds (don't dare try to run it)

sandsmark committed:
it runs, and looks completely wrong

sandsmark committed:
more of the rendering working

sandsmark committed:
clang tidy complained

sandsmark committed:
more minor cleanup

sandsmark committed:
fix animations

sandsmark committed:
fix not exiting when closing window

sandsmark committed:
fix width/height issue

sandsmark committed:
fix strings

sandsmark committed:
throw out miniwin again

sandsmark committed:
cleanup

sandsmark committed:
fix buggy calculation

sandsmark committed:
fix darkness overlay

sandsmark committed:
translate some variables and comments

sandsmark committed:
more translations

sandsmark committed:
start cleaning up renderer

sandsmark committed:
it works?

sandsmark committed:
some cleanup

sandsmark committed:
minor cleanup

sandsmark committed:
fix initialization on start

sandsmark committed:
fix credits

sandsmark committed:
minor cleanup

sandsmark committed:
start embedding resources into binary when building

sandsmark committed:
embed wav files as well

sandsmark committed:
load embedded wavs

sandsmark committed:
allow running just the exe

sandsmark committed:
better?

sandsmark committed:
fix text, ish

sandsmark committed:
blah can't be bothered to fix properly

sandsmark committed:
fix intro

sandsmark committed:
fix strings

sandsmark committed:
fix treasure map

sandsmark committed:
fix animation speed stuff

sandsmark committed:
fix naming

sandsmark committed:
tweak walking animation speed

sandsmark committed:
'windowed fullscreen' by default

sandsmark committed:
update appveyor config

sandsmark committed:
drop unnecessary dep

sandsmark committed:
fix build with msvc maybe

sandsmark committed:
maybe fix msvc v2

sandsmark committed:
maybe fix msvc v3

sandsmark committed:
debugging windows

sandsmark committed:
maybe this fixes wintendo

sandsmark committed:
more debugging of wintendo

sandsmark committed:
ok, maybe it was msvc that was the problem

sandsmark committed:
try to convert to indexed bitmap, maybe SFML handles that on wintendo

sandsmark committed:
change the load ordering to make sure there isn't some garbage static internal state in SFML

sandsmark committed:
sfml is garbage, have to use a PNG to get it to load on windows apparently

sandsmark committed:
avoid some out of bounds

sandsmark committed:
Revert "try to convert to indexed bitmap, maybe SFML handles that on wintendo"

This reverts commit 760ce80af24a4f24ff0c61c093d65a02f07815f0.

sandsmark committed:
goddamn sfml and wintendo

sandsmark committed:
finally fix wintendo I hope

sandsmark committed:
Fix some strings

sandsmark committed:
add developer shortcut for debugging credits

sandsmark committed:
I think the credits are supposed to look something like this

sandsmark committed:
fix water graphics without animation

sandsmark committed:
fix some naming conflicts

sandsmark committed:
ensuring more not colliding

sandsmark committed:
more disambugiationasdf (hard spelling)

sandsmark committed:
continuing the name collision avoidance

sandsmark committed:
so many ambigouisdsoifsdf things

sandsmark committed:
I think that's all, otherwise I need new regexes

sandsmark committed:
ok, missed a couple

sandsmark committed:
fix mouse coordinates being worng

sandsmark committed:
more cleanup

sandsmark committed:
more translations and disambiguation of enums

sandsmark committed:
see if we can fix msvc

sandsmark committed:
fix ambigious sprite enum references once and for all

sandsmark committed:
missing install target

eXpl0it3r commented:
Haven't really commented on this so far, as you've been making a lot of updates still.

Awesome work! 🙂 

Obviously it's kind of hard to review this huge PR, so I wonder:

- What are the main changes you've been doing?
- What are your future plans?

sandsmark commented:
> Obviously it's kind of hard to review this huge PR, so I wonder:

Yeah, I went a bit overboard. :-P

I guess the easiest is just to review the new code as is or something.

> * What are the main changes you've been doing?

Basically just clean everything up. Translated most of the code (and a lot of the comments) from german to english, porting all the graphics stuff from directx to sfml, replaced a lot of windows-specific APIs with cross platform stuff, replaced a lot of defines with enums, cleaning invalid memory stuff, fixed english text in the UI, replaced the loading of game assets (graphics and sound) from the directory the binary is run from with baking them into the executable itself (so you just need the executable to run it, i. e. standalone).

And just a lot of cleaning up the code in general.


> * What are your future plans?

Replacing all the global state with proper classes is probably the next thing I want to do. And in general just making the codebase "modern" and a bit more future proof. I like the game. :-)


FWIW, I made a pkgbuild for archlinux as well and put it on the AUR: https://aur.archlinux.org/packages/schiffbruch-git

sandsmark committed:
more descriptive and less direct translation variable name

sandsmark committed:
a bit better API for the renderer

sandsmark committed:
replace c-style char array with std::string for safety

SimonHeimberg commented:
I was able to compile with cmake on linux after I fixed above problem. Actually this is the same problem as is reported by [AppVeyor](https://ci.appveyor.com/project/eXpl0it3r/schiffbruch/builds/38320071/messages). A big success for me, since I am not used to compiling.

The game is playable but has some bugs:
- [ ] when starting a new game, I still have the material from the previous game. Work around: quit the game, delete save.date (in the current working directory), start the game
- [ ] there is no mark on the map
- [ ] disabling water animation hides fields
- [ ] numbers (after night when sleeping outside) are outside paper and stay
- [ ] start animation often does not play properly on game start. Moving the mouse shows it. It does more likely when starting a new game.
- [ ] when landing with the boat in an edge of the corner, the boat sometimes lands on the wrong edge and a tree is removed or even the river delta (beach with river)
- [ ] man walked once out over the water, before leaving the game (by ESC or btn)

thanks for porting this game to all involved
